src/lib/assetBridger/l1l3Bridger.ts

```diff
@@ -0,0 +1,1141 @@
+    l3Provider: Provider
+  ): Promise<Required<Pick<TransactionRequest, 'to' | 'data' | 'value'>>> {
+    await this._checkL1Network(l1Signer)
+    await this._checkL2Network(l2Provider)
+    await this._checkL3Network(l3Provider)
+
+    const gasParams = await this._populateGasParams(
+      params,
+      l1Signer.provider!,
+      l2Provider,
+      l3Provider
+    )
+
+    const teleporter = L1Teleporter__factory.connect(
+      this.teleporterAddresses.l1Teleporter,
+      l1Signer
+    )
+
+    const calldata = teleporter.interface.encodeFunctionData('teleport', [
+      {
+        l1Token: params.erc20L1Address,
+        l1l2Router: this.l2Network.tokenBridge.l1GatewayRouter,
+        l2l3Router: this.l3Network.tokenBridge.l1GatewayRouter,
+        to: params.to || (await l1Signer.getAddress()),
+        amount: params.amount,
+        gasParams,
+      },
+    ])
+
+    const adjustedL1GasPrice = this._percentIncrease(
+      params.overrides?.l1GasPrice?.base ||
+        (await l1Signer.provider!.getGasPrice()),
+      params.overrides?.l1GasPrice?.percentIncrease ||
+        this.defaultGasPricePercentIncrease
+    )
+
+    const calculatedGasCosts = await teleporter.calculateRetryableGasCosts(
+      this.l2Network.ethBridge.inbox,
+      adjustedL1GasPrice,
+      gasParams
+    )
+
+    return {
+      to: this.teleporterAddresses.l1Teleporter,
+      data: calldata,
+      value: calculatedGasCosts.total,
+    }
+  }
+
+  /**
+   * Deposit tokens to L3 through the L1Teleporter contract.
+   */
+  public async deposit(
+    params: Erc20DepositRequestParams,
+    l1Signer: Signer,
+    l2Provider: Provider,
+    l3Provider: Provider
+  ): Promise<L1ContractCallTransaction> {
+    const txRequest = await this.getDepositRequest(
+      params,
+      l1Signer,
+      l2Provider,
+      l3Provider
+    )
+
+    return L1TransactionReceipt.monkeyPatchContractCallWait(
+      await l1Signer.sendTransaction(txRequest)
+    )
+  }
+
+  /**
+   * Get the status of a deposit given an L1 tx receipt. See `Erc20DepositStatus` interface for more info on the return type.
+   *
+   * Note: This function does not verify that the tx is actually a deposit tx.
+   */
+  public async getDepositStatus(
+    depositTxReceipt: L1ContractCallTransactionReceipt,
+    l2Provider: JsonRpcProvider,
+    l3Provider: Provider
+  ): Promise<Erc20DepositStatus> {
+    const baseStatus = await this._getDepositStatus(
+      depositTxReceipt,
+      l2Provider,
+      l3Provider
+    )
+
+    const secondStepRetryableRedeem = await (
+      await depositTxReceipt.getL1ToL2Messages(l2Provider)
+    )[1].getSuccessfulRedeem()
+
+    return {
+      ...baseStatus,
+      retryableL2ForwarderCall: secondStepRetryableRedeem,
+    }
+  }
+}
+
+/**
+ * Bridge ERC20 tokens from L1 to L3 directly through the token bridge, using a relayer on L2 to call the L2Forwarder contract.
+ */
+export class RelayedErc20L1L3Bridger extends BaseErc20L1L3Bridger {
```

Comment:
I don't feel like having this class in the sdk, it add complexity to the sdk while the relayer feature is unlikely to be used by default. Would prefer to simply having this option documented but not implemented in the sdk (maybe create a tutorial instead).  One risk being if the deposit method crashed due to whatever reason the relayer info might be lost which will lead to fund loss as mentioned. 
If we were to keep this, an option can be appending the relayer config to the deposit transaction to store that info onchain at the expense of increased calldata cost.

Reply:
i really like the idea of appending the relayer info to calldata. i agree that it's a good bit of extra complexity, but I also think it's valuable to have in the sdk given the cost savings.

appending the calldata could also make the api simpler because you wouldn't really need to return / provide relayerinfo anymore, just pass a tx hash or receipt around.

Reply:
if there are no risk of losing fund I think that's fine

Reply:
5d7388a994fca9a0dc6aeff1ad70d8d0235a005c